libs/matdbg/src/JsonWriter.cpp

```diff
@@ -84,11 +85,27 @@ static bool printParametersInfo(ostream& json, const ChunkContainer& container)
 static void printShaderInfo(ostream& json, const std::vector<ShaderInfo>& info) {
     for (uint64_t i = 0; i < info.size(); ++i) {
         const auto& item = info[i];
+        string variantString = "";
+
+        // NOTE: the variant enum is private to Filament so unfortunately we need to decode the bits
+        // using magic constants. The 3-character nomenclature used here is consistent with the
+        // ASCII art seen in the Variant header file and allows the information to fit in a
+        // reasonable amount of space on the page. The HTML file has a legend.
+        if (item.variant) {
+            if (item.variant & 0x01)  variantString += "DIR|";
+            if (item.variant & 0x02)  variantString += "DYN|";
+            if (item.variant & 0x04)  variantString += "SRE|";
+            if (item.variant & 0x08)  variantString += "SKN|";
```

Comment:
You could include #include <private/filament/Variant.h>, it'll be better than tracking down hard coded constants.